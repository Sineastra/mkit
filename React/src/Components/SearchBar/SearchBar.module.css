.form {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: initial;
  align-content: initial;
  white-space: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  height: 100%;
}
.form > .input {
  width: 70%;
  height: 100%;
  border-radius: 5px;
  text-indent: 10px;
}
.form > button {
  width: 25%;
  border-radius: 5px;
  background-color: transparent;
  border: 1px solid var(--theme-text-color);
  color: var(--theme-text-color);
  vertical-align: middle;
  cursor: pointer;
}
.form > button:hover {
  border: 1px solid transparent;
  background: var(--theme-text-color);
  color: var(--theme-main-color_1);
}

/*# sourceMappingURL=SearchBar.module.css.map */
