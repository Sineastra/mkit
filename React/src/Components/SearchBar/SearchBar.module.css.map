{"version":3,"sourceRoot":"","sources":["SearchBar.module.scss","../../_scssPartials/_mixins.scss"],"names":[],"mappings":"AAEA;ECDI;EACA,gBDCqB;ECArB;EACA,aAJgE;EAKhE,eALwF;EDIxF;EACA;EACA;EACA;;AAEA;EACI;EACA;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;EACA","file":"SearchBar.module.css"}