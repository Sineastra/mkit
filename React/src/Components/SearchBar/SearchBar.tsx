import { BaseSyntheticEvent, useState } from "react"
import { useNavigate } from "react-router-dom"
import styles from "./SearchBar.module.scss"


const SearchBar = () => {
	const [searchV, setSearchV] = useState('')
	const navigate = useNavigate()

	const handleChange = (e: BaseSyntheticEvent) => {
		setSearchV(e.target.value)
	}

	const handleSubmit = (e: BaseSyntheticEvent) => {
		e.preventDefault()

		if (searchV.trim() !== '') {
			navigate({
				pathname: '/search',
				search: `?q=${ searchV }`,
			})
			setSearchV('')
		}
	}

	return (
		<form onSubmit={ handleSubmit } className={ styles.form }>
			<input
				type="search"
				placeholder="Search by movie title..."
				onChange={ handleChange }
				value={ searchV }
				className={ styles.input }
			/>
			<button>Search</button>
		</form>
	)
}

export default SearchBar