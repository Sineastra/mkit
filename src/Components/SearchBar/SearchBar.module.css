.form {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: initial;
  align-content: initial;
  white-space: nowrap;
  flex-wrap: nowrap;
}
.form > .input {
  width: 70%;
  height: 100%;
  border-radius: 5px;
  text-indent: 10px;
}
.form > button {
  width: 25%;
  border-radius: 5px;
  background-color: transparent;
  border: 1px solid var(--theme-text-color);
  color: var(--theme-text-color);
  vertical-align: middle;
}

/*# sourceMappingURL=SearchBar.module.css.map */
