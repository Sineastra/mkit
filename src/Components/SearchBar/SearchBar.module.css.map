{"version":3,"sourceRoot":"","sources":["SearchBar.module.scss","../../_scssPartials/mixins/_mixins.scss"],"names":[],"mappings":"AAEA;ECDI;EACA,gBDCqB;ECArB;EACA,aAJgE;EAKhE,eALwF;EDIxF;EACA;;AAEA;EACI;EACA;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;EACA","file":"SearchBar.module.css"}