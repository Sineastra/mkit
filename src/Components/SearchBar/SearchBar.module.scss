@import "../../_scssPartials/mixins/mixins";

.form {
    @include displayFlex(row, space-between);
    white-space: nowrap;
    flex-wrap: nowrap;

    & > .input {
        width: 70%;
        height: 100%;
        border-radius: 5px;
        text-indent: 10px;
    }

    & > button {
        width: 25%;
        border-radius: 5px;
        background-color: transparent;
        border: 1px solid var(--theme-text-color);
        color: var(--theme-text-color);
        vertical-align: middle;
        cursor: pointer;

        &:hover {
            border: 1px solid transparent;
            background: var(--theme-text-color);
            color: var(--theme-main-color_1);
        }
    }
}

