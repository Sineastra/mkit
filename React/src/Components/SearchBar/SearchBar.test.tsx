import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom"
import SearchBar from "./SearchBar"
import { act, render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

const generateData = () => {
	render(
		<BrowserRouter>
			<SearchBar/>
		</BrowserRouter>,
	)

	return screen.getByRole('searchbox') as HTMLInputElement
}

const simulateInputSubmit = async (keyboardSubmit: boolean) => {
	const user = userEvent.setup()
	const searchInput = generateData()
	const searchBtn = screen.getByText('Search')

	await user.type(searchInput, 'a')
	await act(() => {
		return keyboardSubmit ? user.keyboard('{Enter}') : user.click(searchBtn)
	})

	const searchParam = global.location.search

	return { searchParam, searchInput }
}


describe("---> Testing /Components/SearchBar", () => {
	it("snapshot test", () => {
		const tree = renderer.create(
			<BrowserRouter>
				<SearchBar/>
			</BrowserRouter>,
		).toJSON()

		expect(tree).toMatchSnapshot()
	})
	it("should navigate to the proper URL on submit with Enter", async () => {
		const { searchParam } = await simulateInputSubmit(true)

		expect(searchParam).toEqual('?q=a')
	})

	it("should navigate to the proper URL on submit with Click on the Search Icon", async () => {
		const { searchParam } = await simulateInputSubmit(false)

		expect(searchParam).toEqual('?q=a')
	})
	it("should clear input on submit with Enter", async () => {
		const { searchInput } = await simulateInputSubmit(true)

		expect(searchInput.value).toEqual('')
	})
	it("should clear input on submit with Enter", async () => {
		const { searchInput } = await simulateInputSubmit(false)

		expect(searchInput.value).toEqual('')
	})
})