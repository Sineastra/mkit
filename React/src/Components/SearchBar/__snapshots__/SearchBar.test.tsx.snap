// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---> Testing /Components/SearchBar snapshot test 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Search by movie title..."
    type="search"
    value=""
  />
  <button>
    Search
  </button>
</form>
`;
